### PR TITLE
Single Redux store + ReduxRouter for all routes

### PR DESCRIPTION
**Pull request by agilliland**
This is now about 98% of the way towards removing angular from Metabase.

Stuff done here:
- All of the various sections of the app which used their own redux stores have been unified to use a single redux store for the whole app.
- All routes are now handled through ReduxRouter
- The auth pages plus a couple other remaining angular templates have all been moved over to react + redux
- A bunch of unused angular code has been stripped out, including all of the *.module.js files

At this point pretty much all of the angular code is in 4 files: `app.js`, `controllers.js`, `directives.js`, and `services.js`

The biggest outstanding pieces of work left are 1) replace the old angular api client code with something new and 2) rip out the angular bootstrapping + routing code which is still used as a bridge to redux.

TODO:
- [x] `null is not an object (evaluating 'n.id')` right after logging in ( https://www.dropbox.com/s/silj78p0cm0sluj/Screenshot%202016-06-29%2017.40.50.png?dl=0 )
- [x] Adding unsaved card to dashboard broken
- [x] Exception when creating dashboard
- [x] Immediately marks saved SQL card as 'new card' upon loading

### CODE REVIEW

File: frontend/src/metabase/app.js [hunk 1]
```diff
@@ -14,70 +14,108 @@ import 'angular-http-auth';
...
-import "./icons";
 
-import "./auth/auth.module";
-import "./home/home.module";
-import "./pulse/pulse.module";
-import "./query_builder/query_builder.module"
-import "./setup/setup.module";
-import "./user/user.module";
+import React from "react";
+//import { render } from "react-dom";
```
Comment (kdoh) [inline frontend/src/metabase/app.js]: An unused import appears!

File: frontend/src/metabase/app.js [hunk 2]
```diff
@@ -14,70 +14,108 @@ import 'angular-http-auth';
...
+import * as home from "metabase/home/reducers";
+import labels from "metabase/questions/labels";
+import metadata from "metabase/dashboard/metadata";
+import * as people from "metabase/admin/people/reducers";
+import * as pulse from "metabase/pulse/reducers";
+import * as qb from "metabase/query_builder/reducers";
+import questions from "metabase/questions/questions";
+import settings from "metabase/admin/settings/settings";
+import * as setup from "metabase/setup/reducers";
+import undo from "metabase/questions/undo";
```
Comment (kdoh) [inline frontend/src/metabase/app.js]: there are quite a few imports here. my instinct would be to try and sort or group them somehow to make it a bit easier to grok. definitely not blocking but would be quite useful.

**Comment by kdoh** [comment]
@agilliland Auth routes should be displaying properly now. Still need to figure out why the bridge is suddenly rendering weird.

**Comment by agilliland** [comment]
The bridge may be because I had to hack at the html a little to get react to accept it.  react wouldn't allow for any namespacing or things it felt were redundant attributes.  I thought everything was safe to remove, but maybe not?

File: frontend/src/metabase/app.js (hunk 1 above)
Comment (agilliland) [inline frontend/src/metabase/app.js]: Yeah, that was left there from an initial attempt to completely remove angular and bootstrap the app directly using just redux.  I backed off that approach and so i just commented out a few things so it was easy to restore when the time came.

**Comment by kdoh** [comment]
@agilliland  image

**Comment by agilliland** [comment]
yeah, like that.

**Comment by tlrobinson** [comment]
FWIW I tried converting the auth pages a few months ago and also ran into issues with the SVG graphics displaying correctly.

**Comment by kdoh** [comment]
Yeah I think React has a certain "Kaiju must destroy bridge" setting when it comes to certain SVG properties. Looking into that now.

**Comment by tlrobinson** [comment]
_fingers crossed_

**Pull request merged**